docs/decisions/0063-vector-store-list-keys.md

```diff
@@ -0,0 +1,118 @@
+    IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<TKey> keys, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    ...
+    Task<IEnumerable<TKey>> ListKeysAsync(CancellationToken cancellationToken = default);
+}
+```
+Pros:
+- This is a simple and straightforward way to add the functionality
+Cons:
+- It is a breaking change (at least for csharp, python can do a default implementation that returns None)
+
+## Option 2: add a new method to a new separate interface
+
+We can add a separate interface to list all keys in the vector store. This method will return a list of keys.
+
+```csharp
+public interface IVectorStoreRecordCollectionListKeys<TKey, TRecord> : IVectorStoreRecordCollection<TKey, TRecord>
+{
+    Task<IEnumerable<TKey>> ListKeysAsync(CancellationToken cancellationToken = default);
+}
+```
+Pros:
+- This is a non-breaking change, each vector store record collection must choose to implement this interface.
+
+Cons:
+- This is a more complex way to add the functionality
+
+## Option 3: alter the behavior of the Get/GetBatch methods to allow getting all records (key=None or similar)
+
+This options alters the behavior of the Get methods to allow getting all records, this can be done by passing a special key value (e.g. None) to the Get method.
+
+```csharp
+
+public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch<TRecord>
+{
+    Task<TRecord?> GetAsync(TKey? key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<TKey>? keys, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+
+```
+Pros:
+- This allows the other options to be used in concert with getting all records
+- This is a non-breaking change
+Cons:
+- This returns the full records instead of just the keys, which can be useful in some cases, but can be overhead in others.
+- It is not immediately clear that passing a special key value will return all records, a dedicated method would be more clear.
+
+## Options 3b: alter the behavior of the Get/GetBatch methods to allow getting all keys
+
+This options alters the behavior of the Get methods to allow getting all keys, this can be done by passing a special key value (e.g. None) to the Get method, combined with a change in the GetRecordOptions, to specify Keys only.
+
+```csharp
+public interface IVectorStoreRecordCollection<TKey, TRecord> : IVectorizedSearch<TRecord>
+{
+    Task<TRecord?> GetAsync(TKey? key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<TKey>? keys, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+public class GetRecordOptions
+{
+    public bool IncludeVectors { get; init; } = false;
+    public bool KeysOnly { get; init; } = false;
```

Comment:
As described in one of the previous comments, accepting a collection of field/column names to return should be more extensible approach, since it will be possible to specify just a key property name as well as key together with other properties that are required in specific case.